addon/serializer.js

```diff
@@ -116,8 +114,7 @@ export default JSONAPISerializer.extend({
         meta[key] = payload[key];
         delete payload[key];
       });
-    }
-    if (!isSingle) {
+
       meta = extractLinksIntoMeta(payload, meta);
```

Comment:
the `extractLinksIntoMeta` methods took the existing meta object and added fields onto it.
The method naming was off. If the previous meta would be overwritten the test in https://github.com/201-created/ember-data-hal-9000/blob/master/tests/unit/models/meta-test.js#L66 would fail.

I did some cleanup and changed it to `extractLinks` that returns the extracted link object. I also cleaned up some other minor inconsistencies.

There could be a potential bug in case of a payload that uses an existing `meta.links` key. That would be overwritten by the extracted links.

Should we detect this case and warn the user?